### PR TITLE
Documentation for WireGuard component and sensors

### PR DESCRIPTION
**Pull request by droscy**
## Description:

This is the documentation for WireGuard Component and its two sensors: WireGuard Status and WireGuard Latest Handshake.

It takes into account all the features in PR lhoracek/esphome#9 that will be merged in esphome/esphome#4256 as soon as possible.

I have created a custom logo for the index page because the official logo is a registered trademark and cannot be used for custom implementation.

**Related issue (if applicable):**
- original feature request esphome/feature-requests#1444

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4256

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  

  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by droscy** [comment]
This is not stale, the doc is still valid.

File: .gitignore
Comment (jesserockz) [inline .gitignore:1]: Please dont make changes unrelated to wireguard

File: Makefile
Comment (jesserockz) [inline Makefile:1]: Please dont make changes unrelated to wireguard

**Review by jesserockz — CHANGES_REQUESTED** [review]
Docs for parent/child sensors actually mostly live on the same page (the parent page). There are a few exmaples of this, but search for Component/Hub in the docs codebase.

There can still be multiple index links to the same page.

File: Makefile
Comment (droscy) [reply Makefile:1]: My mistake, sorry, I'll revert it.

File: .gitignore
Comment (droscy) [reply .gitignore:1]: I have excluded Eclipse project files and of course I can revert it. But could you please exclude them on your next commit to `next` branch?

**Comment by droscy** [comment]
> There are a few exmaples of this, but search for Component/Hub in the docs codebase.
> There can still be multiple index links to the same page.

@jesserockz I am not sure to fully understand what you asked... should I simply add a "Component/Hub" section at the same level of "Sensor" section?

**Comment by nagyrobi** [comment]
Put all wireguard-related doc in one rst file. No separate files for parent component and child sensors.

**Comment by droscy** [comment]
It is already in a single page, everything is here https://deploy-preview-2948--esphome.netlify.app/components/wireguard

**Comment by nagyrobi** [comment]
This PR still shows that there are several other files involved in the change. Look at the _Files Changed_ tab.

There are also conflicts to be fixed. These prevent the merge. I suggest to either rebase, or close and resubmit the PR freshly.

**Comment by droscy** [comment]
> This PR still shows that there are several other files involved in the change. Look at the _Files Changed_ tab.

Ops, I left them there. I'll remove them.

> There are also conflicts to be fixed.

Because the last commit here was 2 months ago...

**Review by jesserockz — APPROVED** [review]

**Pull request merged**